packages/react-reconciler/src/__tests__/ReactUse-test.js

```diff
@@ -1915,4 +1917,39 @@ describe('ReactUse', () => {
     assertLog(['Hi', 'World']);
     expect(root).toMatchRenderedOutput(<div>Hi World</div>);
   });
+
+  it('regression: does not get stuck in pending state after `use` suspends', async () => {
+    // This is a regression test. The root cause was an issue where we failed to
+    // switch from the "re-render" dispatcher back to the "update" dispatcher
+    // after a `use` suspends and triggers a replay.
```

Comment:
Now it also makes sense why it didn't get stuck if we called `useTransition` before `use`. Thanks for the explainer.